# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: modules/libpref/Preferences.cpp
# Commit: e29b15980da2
# Full Hash: e29b15980da270931b3c798b583687f56a6506a6
# Author: Doug Thayer <dothayer@mozilla.com>
# Date: 2020-07-08 09:42:17
# Regressor Bug: 1627075
# File Overlap Count: 1
# Description:
#   Bug 1627075 - Route Omnijar requests through StartupCache r=froydnj
#   
#   This should be a relatively straightforward patch. Essentially, we implement
#   a wrapper class (and friends) around nsZipArchive (and friends), which transparently
#   caches entries from the underlying zip archive in the StartupCache. This will break
# ==============================================================================

diff -r eb5265addd5e -r e29b15980da2 modules/libpref/Preferences.cpp
--- a/modules/libpref/Preferences.cpp	Tue Jul 07 17:03:28 2020 +0000
+++ b/modules/libpref/Preferences.cpp	Tue Jul 07 17:04:27 2020 +0000
@@ -82,7 +82,6 @@
 #include "nsXPCOMCID.h"
 #include "nsXPCOM.h"
 #include "nsXULAppAPI.h"
-#include "nsZipArchive.h"
 #include "plbase64.h"
 #include "PLDHashTable.h"
 #include "plstr.h"
@@ -4312,7 +4311,7 @@
   return rv;
 }
 
-static nsresult pref_ReadPrefFromJar(nsZipArchive* aJarReader,
+static nsresult pref_ReadPrefFromJar(CacheAwareZipReader* aJarReader,
                                      const char* aName) {
   TimeStamp startTime = TimeStamp::Now();
 
@@ -4329,8 +4328,8 @@
   return NS_OK;
 }
 
-static nsresult pref_ReadDefaultPrefs(const RefPtr<nsZipArchive> jarReader,
-                                      const char* path) {
+static nsresult pref_ReadDefaultPrefs(
+    const RefPtr<CacheAwareZipReader>& jarReader, const char* path) {
   UniquePtr<nsZipFind> find;
   nsTArray<nsCString> prefEntries;
   const char* entryName;
@@ -4509,7 +4508,7 @@
   const char* entryName;
   uint16_t entryNameLen;
 
-  RefPtr<nsZipArchive> jarReader = Omnijar::GetReader(Omnijar::GRE);
+  RefPtr<CacheAwareZipReader> jarReader = Omnijar::GetReader(Omnijar::GRE);
   if (jarReader) {
 #ifdef MOZ_WIDGET_ANDROID
     // Try to load an architecture-specific greprefs.js first. This will be
@@ -4591,7 +4590,7 @@
 
   // Load jar:$app/omni.jar!/defaults/preferences/*.js
   // or jar:$gre/omni.jar!/defaults/preferences/*.js.
-  RefPtr<nsZipArchive> appJarReader = Omnijar::GetReader(Omnijar::APP);
+  RefPtr<CacheAwareZipReader> appJarReader = Omnijar::GetReader(Omnijar::APP);
 
   // GetReader(Omnijar::APP) returns null when `$app == $gre`, in
   // which case we look for app-specific default preferences in $gre.